### ISSUE TITLE
Thanks for sharing. Why use 2 viewpoints during trainging and concat different viewpoint.

### ISSUE THREAD

**Issue by zhiwenfan** [ShichenLiu/SoftRas#36]
Thanks for sharing this great work. There is some confusions with this code. As you mentioned in paper, you are working on single view reconstruction but the code use 2 viewpoints. 
Besides, in model.py, you do concat  viewpoints = torch.cat((viewpoint_a, viewpoint_a, viewpoint_b, viewpoint_b), dim=0) and  vertices = torch.cat((vertices, vertices), dim=0). What's the purposes of such operation? 
Wish to achieve your reply! Thanks.

**Comment by NotAnyMike** [comment]
They are not sharing the whole project. Specifically the renderer is there but the examples are different. Those examples are not the ones described in the paper.

**Comment by ShichenLiu** [comment]
This is for ensuring cross-view consistency. Otherwise, the model will be overfitting to a single silhouette.
